pkg/deploy/generator/scripts/rpVMSS.sh

```diff
@@ -581,7 +581,11 @@ if [ -f \$NEW_CERT_FILE ]; then
   else
     sed -i -ne '1,/END CERTIFICATE/ p' \$NEW_CERT_FILE
   fi
-  if ! diff $NEW_CERT_FILE $CURRENT_CERT_FILE >/dev/null 2>&1; then
+
+  NEW_CERT_SN=$(openssl x509 -in $NEW_CERT_FILE -noout -serial | awk -F= '{print $2}')
```

Comment:
A few suggestions.
1. We should wrap command substituions in quotes to prevent globbing
2. Ideally we should be using lowercase variable names for variables that aren't export, e. g. used only here. It's a little nit picky, but helps us clearly see what is intended for local use vs. exported use
3. The variables inside of command substitution should be quoted, to prevent globbing

Reply:
Also if you make use of `test` with double square brackets, you don't have to quote the left variable.  
The right is still expanded, has some good use cases.  
```bash
if [[ $NEW_CERT_SN != "$CURRENT_CERT_SN" ]]; then
    do stuff
fi
```